web-registry/src/components/organisms/MediaForm/MediaForm.tsx

```diff
@@ -0,0 +1,157 @@
+  creditClassId?: string | null;
+  graphData?: ShaclGraphByUriQuery;
+}
+
+/** Formik Context + handlers for legacy and new media */
+export const MediaForm = ({
+  initialValues,
+  graphData,
+  creditClassId,
+  ...props
+}: MediaFormProps): JSX.Element => {
+  const { confirmSave, isEdit } = useProjectEditContext();
+
+  async function handleSubmit(
+    values: MediaValues,
+    { setSubmitting, setTouched }: FormikHelpers<MediaValues>,
+  ): Promise<void> {
+    setSubmitting(true);
+    try {
+      await props.submit(values);
+      setSubmitting(false);
+      setTouched({}); // reset to untouched
+      if (isEdit && confirmSave) confirmSave();
+    } catch (e) {
+      setSubmitting(false);
+    }
+  }
+
+  const handleValidate = async (values: MediaValues): Promise<MediaErrors> => {
+    const errors: MediaErrors = {};
+    if (graphData?.shaclGraphByUri?.graph) {
+      const projectPageData = {
+        ...getProjectPageBaseData(creditClassId),
+        ...values,
+      };
+      const report = await validate(
+        graphData.shaclGraphByUri.graph,
+        projectPageData,
+        'http://regen.network/ProjectPageMediaGroup',
+      );
+      for (const result of report.results) {
+        const path: string = result.path?.value;
+        let compactedPath: keyof MediaValues | undefined;
+        if (path) {
+          compactedPath = getCompactedPath(path) as
+            | keyof MediaValues
+            | undefined;
+        }
+        // Legacy validation logic
+        if (!isSimpleErrors(errors, creditClassId)) {
+          if (path) {
+            if (compactedPath === 'regen:previewPhoto') {
+              errors[compactedPath] = { '@value': requiredMessage };
+            } else {
+              // for gallery photos, display general error message below "Gallery Photos" section
+              errors['regen:galleryPhotos'] = 'You must add 4 photos';
+            }
+          } else {
+            // sh:or constraint not satisfied on regen:landStewardPhoto/regen:videoURL
+            errors['regen:landStewardPhoto'] = {
+              '@value': requiredMessage,
+            };
+          }
+        } else {
+          // Simple validation logic
+          if (compactedPath) {
+            errors[compactedPath] = { '@value': requiredMessage };
+          }
+        }
+      }
+    }
+    return errors;
+  };
+
+  return (
+    <OnBoardingCard>
+      <Formik
+        enableReinitialize
+        validateOnMount
+        initialValues={initialValues}
+        validate={handleValidate}
+        onSubmit={handleSubmit}
+      >
+        {({ submitForm, isValid, isSubmitting, touched }) => (
+          <>
+            {!!creditClassId ? <MediaFormSimple /> : <MediaFormLegacy />}
+            <ProjectPageFooter
+              onSave={submitForm}
+              onNext={props.onNext}
+              onPrev={props.onPrev}
+              saveDisabled={
+                !isValid || isSubmitting || !Object.keys(touched).length
```

Comment:
I would remove the last check so that the user can directly hit "save and submit" without having to touch a field
```suggestion
                !isValid || isSubmitting
```